priv/catalog/CAEFF1.yaml

```diff
@@ -0,0 +1,44 @@
+id: "CAEFF1"
+name: "OS flavor SLES_SAP"
+group: "OS and package versions"
+description: |
+  Operating system vendor is supported
+remediation: |
+  ## Abstract
+  SAPHanaSR is only supported on SUSE Linux Enterprise Server for SAP Applications.
+
+  ## Remediation
+  Please use SUSE Linux Enterprise Server for SAP Applications.
+
+  ## Reference
+  Azure:
+
+    - https://docs.microsoft.com/en-us/azure/virtual-machines/workloads/sap/sap-hana-high-availability
+
+  AWS:
+
+    - https://docs.aws.amazon.com/sap/latest/sap-hana/sap-hana-on-aws-ha-cluster-configuration-on-sles.html
+
+  GCP:
+
+    - https://cloud.google.com/solutions/sap/docs/sap-hana-os-support
+
+  Nutanix:
+
+    - https://documentation.suse.com/sbp/all/single-html/SLES4SAP-hana-sr-guide-PerfOpt-15/#id-planning-the-installation
+
+  SUSE / KVM:
+
+    - https://documentation.suse.com/sbp/all/single-html/SLES4SAP-hana-sr-guide-PerfOpt-15/#id-planning-the-installation
+
+facts:
+  - name: os_flavor
+    gatherer: package_version
+    argument: SLES_SAP-release
+
+expectations:
+  - name: expectations_SLES_SAP
+    expect: |
+        if facts.os_flavor != "" {
```

Comment:
yeah, seems so. See my comment above about wrong expectation of the return value of `package_version`

Reply:
If the package 'SLES_SAP-release' is not installed the gatherer will fail with an error. In all other cases the gatherer will return the package version, so `facts` will be non-empty. So the check should work.

Reply:
I see. What about the the following alternative? 
```yaml
expect: facts.os_flavor != ""
```
it reduces the `if` to a one-liner and makes sure that the expectation evaluation always returns a boolean, which is what the engine expects.
The current expectation might return a `null` value in case `facts.os_flavor` is acutally empty `""`

---
As a note for completeness there's also the possibility to use an `is_empty()` method on the string, and negate it to say _the flavor should not be empty_
```yaml
expect: !facts.os_flavor.is_empty()
```

No strong preferences, if we find useful the usage of `is_empty()`, we'll track an item to document it.

Reply:
i'd go with the proposed 
```expect: facts.os_flavor != ""```
as it's clearer to read and requires less knowledge of string type rhai functions

Reply:
Thanks for the nice guideline, fixed with new commit.